src/plugins/advanced_settings/public/component_registry/index.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { registerSettingsComponent, getSettingsComponent } from './component_registry';
+import {
+  registerDefaultComponents,
+  PAGE_TITLE_COMPONENT,
+  PAGE_SUBTITLE_COMPONENT,
+  PAGE_FOOTER_COMPONENT,
+} from './default_component_registry';
+
+registerDefaultComponents();
```

Comment:
IMO we should register these in a lifecycle method instead of having them registered as a side effect of importing this file.

In fact, I'd consider removing this file entirely, and constructing the API inside of `plugin.ts` instead since it's pretty straightforward. Then the `registerDefaultComponents` could happen inside `setup`

Reply:
K, will dig in. Was avoiding making more changes but I might as well. I think some of this code should be rewritten to use classes otherwise the statefulness is defined in `component_registry.ts`  which feels kind of weird  - no instance instantiation.